### PR TITLE
Utilize a logging interface

### PR DESCRIPTION
**Pull request by rogerclermont**
Create/use a logging interface, with the option for a user-specified override.  Allows the user to tie into any existing logging infrastructure.

### CODE REVIEW

File: gomigrate.go [hunk 1]
```diff
@@ -34,20 +35,34 @@ type Migrator struct {
 	migrations     map[uint64]*Migration
 }
 
+// Logger is the instance of a StdLogger interface to log events to.
+// By default it is set to send all log messages to stdout,
+// but you can set it to redirect wherever you want by instantiating
+// your own logger.
+var Logger StdLogger = log.New(os.Stdout, "[gomigrate] ", log.LstdFlags)
```
Comment (DavidHuie) [inline gomigrate.go]: I would make this a struct attribute of `Migrator`.

File: gomigrate.go [hunk 2]
```diff
@@ -34,20 +35,34 @@ type Migrator struct {
...
+
+// StdLogger is used to log error messages.
+type StdLogger interface {
```
Comment (DavidHuie) [inline gomigrate.go]: Let's just call this `Logger`.

File: gomigrate.go (hunk 1 above)
Comment (DavidHuie) [inline gomigrate.go]: Let's then set the logger to be the stdout one in `NewMigrator`. We can then allow for a custom logger in a different initializer, like `NewCustomMigrator`.

**Comment by rogerclermont** [comment]
Set up in this manner, the Logger is global and thus available for use anywhere within the gomigrate package, whereas making it a property of Migrator restricts its use to only within Migrator methods.

**Comment by rogerclermont** [comment]
Restating my last comment, the logger was intentionally left in global space so as to make it available throughout the entire go-migrate package.  With that said, do you still desire that the logger be a property of Migrator?

**Comment by DavidHuie** [comment]
Yes, I think creating a logger per migrator gives us the most flexibility.

**Pull request closed without merging**